### ISSUE TITLE
Update Gatsby migration doc to reflect new content behavior

### ISSUE THREAD

**Issue by JusticeMatthew** [withastro/docs#10189]
### 📚 Subject area/topic

guide

### 📋 Page(s) affected (or suggested, for new content)

https://docs.astro.build/en/guides/migrate-to-astro/from-gatsby/#migrating-pages-and-posts

### 📋 Description of content that is out-of-date or incorrect

As mentioned by @bholmesdev in discord, the linked Gatsby migration documentation still states that in Astro all content must be in the `src` directory

This is no longer the case thanks to the content layer

**Comment by sarah11918** [comment]
Indeed! I think many of our "migrate to" probably follow this wording. (Whatever Gatsby says, probably Next and Nuxt say something similar as those are our three "most full" guides).

Would happily accept a PR for someone scanning through for outdated language regarding content collections restrictions and updating for the new reality in v5! 🚀

**Comment by sarah11918** [comment]
the three guides migrating from Gatsby, Next, and Nuxt all probably have this identical section:

Migrating Pages and Posts
In Gatsby, your [pages and posts](https://docs.astro.build/en/basics/astro-pages/) may exist in src/pages/ or outside of src in another folder, like content. In Astro, all your page content must live within src/.

Let's change this to:
Migrating Pages and Posts
In Gatsby, your [pages and posts](/en/basics/astro-pages/) may exist in src/pages/ or outside of src in another folder, like `/content/`. With [content collections](/en/guides/content-collections/), your source content can also exist outside of your Astro project entirely.
